tools/xtce_generation/xtce_generator_template.py

```diff
@@ -1,47 +1,105 @@
+    """
+    description = (
+        "This command line program generates an instrument specific XTCE file."
+        "Example usage: "
+        '--instrument "swapi"'
+        '--file_name "TLM_SWP_20231006-121021.xlsx"'
+        "--packets '"
+        '{"P_SWP_HK": 1184, '
+        '"P_SWP_SCI": 1188, '
+        '"P_SWP_AUT": 1192, '
+        "}'"
+    )
+    instrument_help = (
+        "The instrument to process. Acceptable values are: "
+        f"{imap_data_access.VALID_INSTRUMENTS}"
+    )
+    file_name_help = "Provide file name to write packets to."
+    packets_help = (
+        "Provide packet dictionary using packet_name, and app_id."
+        '{"<packet_name>": <app_id>}'
+    )
+
+    parser = argparse.ArgumentParser(prog="imap_xtce", description=description)
+    parser.add_argument("--instrument", type=str, required=True, help=instrument_help)
+    parser.add_argument("--file_name", type=str, required=True, help=file_name_help)
+    parser.add_argument("--packets", type=str, required=True, help=packets_help)
+
+    args = parser.parse_args()
+
+    return args
+
+
+def _validate_args(args):
+    """
+    Ensure that the instrument is valid.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        An object containing the parsed arguments and their values.
+    """
+    if args.instrument not in imap_data_access.VALID_INSTRUMENTS:
+        raise ValueError(
+            f"{args.instrument} is not in the supported instrument list: "
+            f"{imap_data_access.VALID_INSTRUMENTS}"
+        )
+
+    directory = Path(__file__).parent / args.file_name
+    if not directory.exists():
+        raise FileNotFoundError(f"{args.file_name} does not exist.")
+
+
 def main():
-    """Function used by instrument to generate XTCE. Change values where TODO is."""
-
-    # TODO: change instrument name
-    instrument_name = "<instrument_name>"
-    current_directory = Path(__file__).parent
-    module_path = f"{current_directory}/../../imap_processing"
-    packet_definition_path = f"{module_path}/{instrument_name}/packet_definitions"
-    # TODO: Copy packet definition to tools/xtce_generation/ folder
-    path_to_excel_file = f"{current_directory}/<excel_file_name>"
-
-    # TODO: update packets dictionary with packet name and appId
-    # Eg.
-    # packets = {
-    #     "P_COD_HI_PHA": 1169,
-    #     "P_COD_LO_PHA": 1153,
-    #     "P_COD_LO_NSW_SPECIES_COUNTS": 1157,
-    #     "P_COD_HI_OMNI_SPECIES_COUNTS": 1172,
-    # }
-
-    packets = {
-        "<packet_name>": "<app_id_number>",
-    }
+    """Generate xtce file from CLI information given."""
+
+    # Parse arguments, and validate instrument
+    args = _parse_args()
+    _validate_args(args)
+
+    instrument_name = args.instrument
+    current_directory = Path(__file__).parent.parent.parent
+    module_path = current_directory / "imap_processing"
+    packet_definition_path = module_path / instrument_name / "packet_definitions"
+    path_to_excel_file = (
```

Comment:
Interested to hear if @maxinelasp agrees, but I'd prefer the command argument be the full path to the excel file rather than needing to put the excel file in the xtce_generation folder. I don't think we want to store the excel sheets in the repo so it's an extra step to copy one in there to run this command.

Reply:
Yep, I agree, good call

Reply:
Good stuff! I'll get that implemented, and then request another review.